diff --git a/fs/proc/cmdline.c b/fs/proc/cmdline.c
index e851568..5b78503 100644
--- a/fs/proc/cmdline.c
+++ b/fs/proc/cmdline.c
@@ -4,6 +4,33 @@
 #include <linux/kernel.h>
 #include <linux/proc_fs.h>
 #include <linux/seq_file.h>
+#include <linux/string.h>
+
+static int __init cmdline_hide_bl_lock(void)
+{
+	char *ptr;
+
+	ptr = strstr(saved_command_line, "androidboot.flash.locked=0");
+	if (ptr)
+		ptr[sizeof("androidboot.flash.locked=") - 1] = '1';
+
+	ptr = strstr(saved_command_line, "androidboot.vbmeta.device_state=unlocked");
+	if (ptr) {
+		size_t off = sizeof("androidboot.vbmeta.device_state=") - 1;
+		memmove(ptr + off + 6, ptr + off + 8, strlen(ptr + off + 8) + 1);
+		memcpy(ptr + off, "locked", 6);
+	}
+
+	ptr = strstr(saved_command_line, "androidboot.verifiedbootstate=orange");
+	if (ptr) {
+		size_t off = sizeof("androidboot.verifiedbootstate=") - 1;
+		memmove(ptr + off + 5, ptr + off + 6, strlen(ptr + off + 6) + 1);
+		memcpy(ptr + off, "green", 5);
+	}
+
+	return 0;
+}
+arch_initcall(cmdline_hide_bl_lock);
 
 static int cmdline_proc_show(struct seq_file *m, void *v)
 {
